arch/avr/src/common/avr_assert.c

```diff
@@ -23,102 +23,12 @@
  ****************************************************************************/
 
 #include <nuttx/config.h>
-
-#include <stdint.h>
-#include <stdlib.h>
-#include <assert.h>
-#include <debug.h>
```

Comment:
why the inclusion is different? please ensure all xxx-assert.c contain the same header files.

Reply:
This is because some arch(such as arm64_assert.c) put dump logic in xxx_assert.c, while some (such as mips_assert.c) put dump logic in another xxxx_dumpstate.c.
Unless we move all dump state logic into a common file, include files cannot be removed.
As I have mentioned before, moving all dump state logic together will be in another patch (Lots of work). So how about we focus on common logic other than dump state in this patch?